src/ai/backend/manager/actions/action.py

```diff
@@ -27,6 +29,21 @@ class BaseActionResult(ABC):
     def entity_id(self) -> Optional[str]:
         raise NotImplementedError
 
+    @abstractmethod
+    def status(self) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def description(self) -> Optional[str]:
+        raise NotImplementedError
+
+    def to_bgtask_result(self) -> DispatchResult:
+        """
+        Implement this method to convert the Result to DispatchResult.
+        You should implment this method for running the action as bgtask.
+        """
+        raise NotImplementedError
```

Comment:
Since all actions are not always used in a bgtask, it makes sense to implement from in the bgtask.